test/e2e/e2e_test.cpp

```diff
@@ -122,6 +122,36 @@ TEST_F(E2ETest, MultiHomeMutliPartitionTxn) {
   }
 }
 
+TEST_F(E2ETest, RemasterTxn) {
+  for (size_t i = 0; i < NUM_MACHINES; i++) {
+    // auto txn = MakeTransaction({"A", "X"} /* read_set */, {}  /* write_set */);
+
+    auto remaster_txn = MakeTransaction(
+        {}, /* read_set */
+        {"A"},  /* write_set */
+        "", /* code */
+        {}, /* master metadata */
+        MakeMachineId("0:0") /* coordinating server */,
+        1 /* new master */);
+
+    test_slogs[i]->SendTxn(remaster_txn);
+    auto remaster_txn_resp = test_slogs[i]->RecvTxnResult();
+    ASSERT_EQ(TransactionStatus::COMMITTED, remaster_txn_resp.status());
+    ASSERT_EQ(TransactionType::SINGLE_HOME, remaster_txn_resp.internal().type());
+
+    auto txn = MakeTransaction({"A", "X"} /* read_set */, {}  /* write_set */);
+
+    // Note: this relies on metadata being updated synchronously, or the txn may abort.
+    // May break in future implementations
```

Comment:
Maybe I'm missing a better way to do e2e remaster testing. This won't work for testing an abort, since incorrect master metadata would be overwritten...

Reply:
I think you can use `Data()` in test slog to change the stored metadata on one replica then send a non-home txn to forward it to a different replica.

Reply:
I think this causes a data race- when the transaction is forwarded back to that replica with the incorrect metadata, it will commit the transaction while the others abort. Maybe we could make the forwarder avoid overwriting existing master metadata

Reply:
agreed